tm2unixtime.c

```diff
@@ -117,29 +118,29 @@ static int do_range_limit_days(timelib_sll *y, timelib_sll *m, timelib_sll *d)
 	do_range_limit(1, 13, 12, m, y);
 
 	leapyear = timelib_is_leap(*y);
-	days_this_month = leapyear ? days_in_month_leap[*m] : days_in_month[*m];
-	last_month = (*m) - 1;
-
-	if (last_month < 1) {
-		last_month += 12;
-		last_year = (*y) - 1;
-	} else {
-		last_year = (*y);
-	}
-	leapyear = timelib_is_leap(last_year);
-	days_last_month = leapyear ? days_in_month_leap[last_month] : days_in_month[last_month];
+	days_per_month_current_year = leapyear ? days_in_month_leap : days_in_month;
+
+	while (*d <= 0 && *m > 0) {
+		last_month = (*m) - 1;
+		if (last_month < 1) {
+			last_month += 12;
+			last_year = (*y) - 1;
+		} else {
+			last_year = (*y);
+		}
+		leapyear = timelib_is_leap(last_year);
+		days_last_month = leapyear ? days_in_month_leap[last_month] : days_in_month[last_month];
```

Comment:
Would `days_last_month` not be better named as `days_in_previous_months` as it is no longer about just days in a single month?